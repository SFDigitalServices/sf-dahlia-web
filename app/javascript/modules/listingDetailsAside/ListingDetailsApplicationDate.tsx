import React from "react"
import { RailsListing } from "../listings/SharedHelpers"
import { ApplicationStatus, ApplicationStatusType, Icon, t } from "@bloom-housing/ui-components"
import { localizedFormat } from "../../util/languageUtil"
import dayjs from "dayjs"
import { LISTING_TYPE_FIRST_COME_FIRST_SERVED } from "../constants"
import { useFeatureFlag } from "../../hooks/useFeatureFlag"
import { isFcfsSalesListing, isOpen } from "../../util/listingUtil"
import bloomTheme from "../../../../tailwind.config"
import { Message } from "@bloom-housing/ui-seeds"

export interface ListingDetailsApplicationDateProps {
  listing: RailsListing
}

export const formatTime = (time: string) => {
  const formattedTime = dayjs(time).format("h:mm")
  const hour = Number(dayjs(time).format("H"))
  // \u00A0 is a non-breaking space
  const suffix = hour >= 12 ? "\u00A0PM" : "\u00A0AM"
  return `${formattedTime}${suffix}`
}

const StatusMessage = ({ isClosed, message }: { isClosed: boolean; message: string }) => {
  return (
    <Message
      fullwidth
      className="justify-start leading-5"
      variant={isClosed ? "alert" : "primary"}
      customIcon={<Icon fill={isClosed ? "red-700" : ""} symbol="clock" size="medium" />}
    >
      {message}
    </Message>
  )
}

const getStatusLottery = (listing: RailsListing) => {
  const isClosed = !isOpen(listing)
  const datetime = listing.Application_Due_Date
  return (
    <StatusMessage
      isClosed={isClosed}
      message={t(
        isClosed
          ? "listingDetails.applicationsClosed.withDateTime"
          : "listingDetails.applicationsDeadline.withDateTime",
        {
          date: localizedFormat(datetime, "LL"),
          time: formatTime(datetime),
        }
      )}
    />
  )
}

const getStatusFcfs = (listing: RailsListing) => {
  const isClosed = !listing.Accepting_Online_Applications
  const datetime = listing.Application_Start_Date_Time
  const showDateTime = new Date(listing.Application_Start_Date_Time) > new Date()

  const getMessage = () => {
    if (isClosed) {
      return t("listingDetails.applicationsClosed")
    } else {
      return showDateTime
        ? t("listingDetails.applicationsOpen.withDateTime", {
            date: localizedFormat(datetime, "LL"),
            time: formatTime(datetime),
          })
        : t("listingDetails.applicationsOpen")
    }
  }

  return <StatusMessage isClosed={isClosed} message={getMessage()} />
}

const ListingDetailsStatus = ({ listing }: { listing: RailsListing }) => {
  return listing.Listing_Type === LISTING_TYPE_FIRST_COME_FIRST_SERVED
    ? getStatusFcfs(listing)
    : getStatusLottery(listing)
}

export const ListingDetailsApplicationDate = ({ listing }: ListingDetailsApplicationDateProps) => {
  const isApplicationOpen = listing && isOpen(listing)

  const { unleashFlag: isSalesFcfsEnabled } = useFeatureFlag("FCFS", false)

  return (
    <div className="w-full mb-8 md:mb-0">
      {isSalesFcfsEnabled && isFcfsSalesListing(listing) ? (
        <ListingDetailsStatus listing={listing} />
      ) : (
        <div className="w-full mb-8 md:mb-0">
          <ApplicationStatus
            className="place-content-center"
            content={t(
              isApplicationOpen
                ? "listingDetails.applicationDeadline.open"
                : "listingDetails.applicationDeadline.closed",
              {
                date: localizedFormat(listing.Application_Due_Date, "ll"),
                time: dayjs(listing.Application_Due_Date).format("h:mm A"),
              }
            )}
            iconColor={!isApplicationOpen && bloomTheme.theme.colors.red["700"]}
            status={isApplicationOpen ? ApplicationStatusType.Open : ApplicationStatusType.Closed}
          />
        </div>
      )}
    </div>
  )
}
