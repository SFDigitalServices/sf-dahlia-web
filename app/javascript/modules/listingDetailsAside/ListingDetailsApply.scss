.fcfs-bmr-how-to-apply {
  padding: var(--seeds-spacer-content) !important;
}

.fcfs-bmr-how-to-apply__list {
  padding-top: var(--seeds-spacer-content) !important;
  padding-bottom: var(--seeds-spacer-content) !important;
}

.fcfs-bmr-how-to-apply {
  h4.text__underline-weighted {
    margin-bottom: 0px;
  }
}
