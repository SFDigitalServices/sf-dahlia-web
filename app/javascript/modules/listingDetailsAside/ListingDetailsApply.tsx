import React, { useState } from "react"
import {
  AppearanceStyleType,
  Button,
  Heading,
  LinkButton,
  MultiLineAddress,
  OrDivider,
  SidebarBlock,
  t,
} from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import {
  acceptingPaperApplications,
  isHabitatListing,
  isOpen,
  isRental,
  isSale,
  paperApplicationURLs,
} from "../../util/listingUtil"
import { localizedFormat, renderInlineMarkup } from "../../util/languageUtil"

export interface ListingDetailsApplyProps {
  listing: RailsListing
}

export const ListingDetailsApply = ({ listing }: ListingDetailsApplyProps) => {
  const [paperApplicationsOpen, setPaperApplicationsOpen] = useState(false)

  if (!isOpen(listing)) return null

  const isListingRental = isRental(listing)

  const acceptingPaperApps = acceptingPaperApplications(listing)

  const ordinalHeader = (ordinal: number, title: string) => {
    return (
      <Heading priority={4} className={"text-gray-900 text-lg -mt-2 mb-3"}>
        <span className={"text-blue-600 mr-2"}>{ordinal}</span>
        {title}
      </Heading>
    )
  }

  const howToApplyBlock = (
    <SidebarBlock title={t("listings.apply.howToApply")}>
      {!isListingRental && (
        <>
          <p className={"mb-4"}>
            {renderInlineMarkup(
              t("listings.apply.fulfillEligibilityRequirements", {
                url: isHabitatListing(listing)
                  ? "https://habitatgsf.org/amber-drive-info/"
                  : "https://sfmohcd.org/homebuyer-program-eligibility",
              })
            )}
          </p>
          <p className={"mb-4"}>{t("listings.apply.eligibilityRequirementDescription")}</p>
        </>
      )}
      <LinkButton
        styleType={AppearanceStyleType.primary}
        className={"w-full"}
        transition={true}
        href={`listings/${listing.listingID}/apply-welcome/intro`}
      >
        {t("label.applyOnline")}
      </LinkButton>
      {process.env.COVID_UPDATE && (
        <div className={"mt-4"}>
          <Heading priority={4} className={"text-base text-gray-800 font-sans"}>
            {t("listings.apply.covidUpdate")}
          </Heading>
          <div className={"text-gray-700 text-base mt-2"}>
            {t("listings.apply.covidUpdateInfo")}
          </div>
        </div>
      )}
      {acceptingPaperApps && (
        <>
          <OrDivider bgColor={"white"} />
          {ordinalHeader(1, t("listings.apply.getAPaperApplication"))}
          {t("listings.apply.paperApplicationsMustBeMailed")}
          <Button
            className={"w-full mt-4"}
            transition={true}
            icon={"arrowDown"}
            iconPlacement={"right"}
            onClick={() => setPaperApplicationsOpen(!paperApplicationsOpen)}
          >
            {t("label.downloadApplication")}
          </Button>
          {paperApplicationsOpen && acceptingPaperApplications && (
            <div className={"flex w-full items-center justify-center flex-col"}>
              {paperApplicationURLs(isListingRental).map((app) => {
                return (
                  <span className={"pt-4"}>
                    <LinkButton
                      href={app.fileURL}
                      unstyled
                      className={"m-0 no-underline"}
                      newTab={true}
                    >
                      {app.languageString}
                    </LinkButton>
                  </span>
                )
              })}
            </div>
          )}
        </>
      )}
    </SidebarBlock>
  )

  const submitPaperApplicationBlocks = (
    <>
      <SidebarBlock className={"bg-blue-200"}>
        {ordinalHeader(2, t("listings.apply.submitAPaperApplication"))}
        {t("listings.apply.includeAnEnvelope")}
      </SidebarBlock>
      <SidebarBlock
        className={"bg-blue-200"}
        styleType={"capsWeighted"}
        title={t("listings.apply.sendByUsMail")}
      >
        <div className={"mb-2 text-gray-900 text-base"}>
          <MultiLineAddress
            address={{
              placeName: listing.Application_Organization,
              street: listing.Application_Street_Address,
              city: listing.Application_City,
              zipCode: listing.Application_Postal_Code,
              state: listing.Application_State,
            }}
          />
        </div>
        {t("listings.apply.applicationsMustBeReceivedByDeadline")}
      </SidebarBlock>
      <SidebarBlock className={"bg-blue-200"}>
        {t("listings.doNotApplyOneAndPaperOrMultiple")}
      </SidebarBlock>
    </>
  )

  const needHelpBlock = (
    <SidebarBlock title={t("listings.apply.needHelp")}>
      {isListingRental && (
        <div className={"mb-4"}>{t("listings.apply.visitAHousingCounselor")}</div>
      )}
      <LinkButton
        transition={true}
        newTab={true}
        href={
          !isListingRental
            ? "https://www.homeownershipsf.org/application-assistance-for-homebuyers/"
            : "/housing-counselors"
        }
        className={"w-full"}
      >
        {isListingRental
          ? t("housingCounselor.findAHousingCounselor")
          : t("listings.apply.visitHomeownershipSf")}
      </LinkButton>
    </SidebarBlock>
  )

  const expectedMoveInDateBlock = (
    <SidebarBlock title={t("listings.expectedMoveinDate")}>
      {localizedFormat(listing.Expected_Move_in_Date, "MMMM YYYY")}
    </SidebarBlock>
  )

  return (
    <>
      {howToApplyBlock}
      {acceptingPaperApps && submitPaperApplicationBlocks}
      {needHelpBlock}
      {isSale(listing) && listing.Expected_Move_in_Date && expectedMoveInDateBlock}
    </>
  )
}
