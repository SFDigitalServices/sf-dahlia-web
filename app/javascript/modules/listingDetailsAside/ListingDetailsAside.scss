.fcfs-aside {
    --card-border-radius: rounded-none;
    --card-background-color: var(--bloom-color-gray-200);
    --card-footer-background-color: var(--bloom-color-gray-200);
    --card-border-width: border-b-1;

    .aside-card-header {
        font-family: var(--seeds-font-alt-sans);
        font-size: var(--seeds-font-size-lg);
        font-weight: var(-seeds-font-weight-semibold);
        line-height: var(--seeds-line-height-6);
    }

    .aside-card-section {
        font-family: var(--seeds-font-sans);
        font-size: var(--seeds-font-size-sm);
        font-weight: var(--seeds-font-weight-regular);
        line-height: var(--seeds-line-height-5);
    }
}
