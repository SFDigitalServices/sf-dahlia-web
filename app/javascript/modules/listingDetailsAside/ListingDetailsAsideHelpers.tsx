import { t } from "@bloom-housing/ui-components"
import React from "react"

export const SeeThisUnitMLS = ({ url }: { url: string }) => {
  return (
    <p className="mt-1">
      <a href={url} target="_blank">
        {t("listings.process.seeThisUnitOnMls")}
      </a>
    </p>
  )
}
