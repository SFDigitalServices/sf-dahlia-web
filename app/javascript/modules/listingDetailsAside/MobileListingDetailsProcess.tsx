import React from "react"
import { ListingDetailItem, t, Mobile } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsInfoSession } from "./ListingDetailsInfoSession"
import { ListingDetailsProcess } from "./ListingDetailsProcess"
import { ListingDetailsLotteryResults } from "../listingDetailsLottery/ListingDetailsLotteryResults"
import { ListingDetailsLotteryInfo } from "../listingDetailsLottery/LotteryDetailsLotteryInfo"
import { ListingDetailsOpenHouses } from "./ListingDetailsOpenHouses"

export interface ListingDetailsSidebarProps {
  listing: RailsListing
  imageSrc: string
  isApplicationOpen: boolean
  whatToEpectContent: string
}

export const MobileListingDetailsProcess = ({
  listing,
  imageSrc,
  isApplicationOpen,
  whatToEpectContent
}: ListingDetailsSidebarProps) => {
  return (
    listing &&
    isApplicationOpen && (
      <Mobile>
        <ListingDetailItem
          imageAlt={""}
          imageSrc={imageSrc}
          title={t("listings.process.header")}
          subtitle={t("listings.process.subheader")}
          hideHeader={true}
          desktopClass="header-hidden"
        >
          <ListingDetailsLotteryInfo listing={listing} />
          <ListingDetailsLotteryResults listing={listing} />
          {isApplicationOpen && <ListingDetailsInfoSession listing={listing} />}
          <ListingDetailsOpenHouses listing={listing} />
          <ListingDetailsProcess listing={listing} isApplicationOpen={isApplicationOpen} whatToExpectContent={whatToEpectContent} />
        </ListingDetailItem>
      </Mobile>
    )
  )
}
