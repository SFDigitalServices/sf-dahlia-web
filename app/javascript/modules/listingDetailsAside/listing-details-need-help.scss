.listing-details-need-help {
  padding: var(--seeds-spacer-content) !important;
  border-color: var(--bloom-color-gray-400) !important;
}
