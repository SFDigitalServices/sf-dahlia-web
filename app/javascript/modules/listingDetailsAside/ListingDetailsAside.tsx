import React from "react"
import { ListingDetailItem, t } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsInfoSession } from "./ListingDetailsInfoSession"
import { ListingDetailsProcess } from "./ListingDetailsProcess"
import { isOpen } from "../../util/listingUtil"
import { ListingDetailsApply } from "./ListingDetailsApply"
import { ListingDetailsApplicationDate } from "./ListingDetailsApplicationDate"
import { ListingDetailsLotteryResults } from "../listingDetailsLottery/ListingDetailsLotteryResults"

export interface ListingDetailsSidebarProps {
  listing: RailsListing
  imageSrc: string
}

export const ListingDetailsAside = ({ listing, imageSrc }: ListingDetailsSidebarProps) => {
  const isApplicationOpen = isOpen(listing)

  return (
    <ListingDetailItem
      imageAlt={""}
      imageSrc={imageSrc}
      title={t("listings.process.header")}
      subtitle={t("listings.process.subheader")}
      hideHeader={true}
      desktopClass="header-hidden"
    >
      <aside className="w-full static md:absolute md:right-0 md:w-1/3 md:top-0 sm:w-2/3 md:ml-2 h-full md:border border-solid bg-white">
        <div className="hidden md:block">
          <ListingDetailsApplicationDate isApplicationOpen={isApplicationOpen} listing={listing} />
          <ListingDetailsLotteryResults listing={listing} />
          {isApplicationOpen && <ListingDetailsInfoSession listing={listing} />}
          <ListingDetailsApply listing={listing} />
          <ListingDetailsProcess listing={listing} />
        </div>
      </aside>
    </ListingDetailItem>
  )
}
