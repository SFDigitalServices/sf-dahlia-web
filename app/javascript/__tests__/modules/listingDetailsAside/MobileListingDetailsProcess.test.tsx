import React from "react"
import renderer from "react-test-renderer"
import { MobileListingDetailsProcess } from "../../../modules/listingDetailsAside/MobileListingDetailsProcess"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"
import { closedRentalListing } from "../../data/RailsRentalListing/listing-rental-closed"

describe("MobileListingDetailsProcess", () => {
  it("does not render if listing is closed", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })
    const tree = renderer
      .create(
        <MobileListingDetailsProcess
          listing={closedRentalListing}
          imageSrc=""
          isApplicationOpen={false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders with listing with open application", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })
    const tree = renderer
      .create(
        <MobileListingDetailsProcess
          listing={openSaleListing}
          imageSrc=""
          isApplicationOpen={true}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
