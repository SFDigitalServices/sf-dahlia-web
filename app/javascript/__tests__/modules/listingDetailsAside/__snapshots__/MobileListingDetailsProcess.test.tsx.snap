// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileListingDetailsProcess does not render if listing is closed 1`] = `null`;

exports[`MobileListingDetailsProcess renders with listing with open application 1`] = `
<li
  className="responsive-content-item header-hidden"
>
  <header
    className="detail-header md:hidden"
  >
    <img
      alt=""
      className="detail-header__image "
      src=""
    />
    <hgroup
      className="detail-header__hgroup"
    >
      <h2
        className="detail-header__title"
      >
        Process
      </h2>
      <span
        className="detail-header__subtitle"
      >
        Important dates and contact information
      </span>
      <span
        className="ui-icon ui-medium"
        data-test-id={null}
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
        >
          <path
            d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
          />
        </svg>
      </span>
    </hgroup>
  </header>
  <div
    className="border-b border-gray-400 md:border-b-0"
  >
    <section
      className="aside-block"
    >
      <h4
        className="text__underline-weighted"
      >
        Information Sessions
      </h4>
      <div
        className="false"
      >
        <p
          className="text text-gray-800 pb-2 flex justify-between items-center"
        >
          <span
            className="inline-block text-tiny uppercase normal-case"
          >
            November 19, 2019
          </span>
          <span
            className="inline-block text-sm font-bold ml-5 font-alt-sans"
          >
            6:00 PM
          </span>
        </p>
        <div
          className="text-tiny text-gray-700 false"
        >
          <div
            className="flex flex-col"
          >
            <span
              className="translate"
            >
              <span>
                MOHCD
              </span>
            </span>
            <span>
              1 South Van Ness Ave, Fl 5, San Francisco
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    className="border-b border-gray-400 md:border-b-0"
  >
    <section
      className="aside-block"
    >
      <h4
        className="text__underline-weighted"
      >
        Open Houses
      </h4>
      <div
        className="pb-3"
      >
        <p
          className="text text-gray-800 pb-2 flex justify-between items-center"
        >
          <span
            className="inline-block text-tiny uppercase normal-case"
          >
            December 3, 2019
          </span>
          <span
            className="inline-block text-sm font-bold ml-5 font-alt-sans"
          >
            2:00 PM - 6:00 PM
          </span>
        </p>
        <div
          className="text-tiny text-gray-700 pb-3"
        >
          <div
            className="flex flex-col"
          />
        </div>
      </div>
      <div
        className="pb-3"
      >
        <p
          className="text text-gray-800 pb-2 flex justify-between items-center"
        >
          <span
            className="inline-block text-tiny uppercase normal-case"
          >
            December 7, 2019
          </span>
          <span
            className="inline-block text-sm font-bold ml-5 font-alt-sans"
          >
            10:00 AM - 12:00 PM
          </span>
        </p>
        <div
          className="text-tiny text-gray-700 pb-3"
        >
          <div
            className="flex flex-col"
          >
            <span
              className="translate"
            >
              <span>
                SFPL
              </span>
            </span>
            <span>
              1 Library Way, San Francisco
            </span>
          </div>
        </div>
      </div>
      <div
        className="false"
      >
        <p
          className="text text-gray-800 pb-2 flex justify-between items-center"
        >
          <span
            className="inline-block text-tiny uppercase normal-case"
          >
            December 8, 2019
          </span>
          
        </p>
        <div
          className="text-tiny text-gray-700 false"
        >
          <div
            className="flex flex-col"
          />
        </div>
      </div>
    </section>
  </div>
  <div
    className="border-b border-gray-400 md:border-b-0"
  >
    <section
      className="aside-block"
    >
      <h4
        className="text__underline-weighted"
      >
        What to Expect
      </h4>
      <div
        className="text-tiny text-gray-750"
      >
        <span>
          Applicants will be contacted by the agent in lottery rank order until vacancies are filled.
        </span>
        <div
          className="mt-2"
        >
          <div>
            <button
              aria-expanded={false}
              className="button is-unstyled m-0 no-underline has-toggle"
              onClick={[Function]}
              type="button"
            >
              Read More
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    className="border-b border-gray-400 md:border-b-0 last:border-b-0"
  >
    <section
      className="aside-block"
    >
      <h4
        className="text__underline-weighted"
      >
        Contact
      </h4>
      <p
        className="text-xl"
      >
        Matthew Burns
      </p>
      <p
        className="text-gray-700"
      >
        Leasing Agent
      </p>
      <p
        className="mt-3"
      >
        <a
          href="tel:Call 415 5552323"
        >
          <span
            className="ui-icon ui-medium pr-2"
            data-test-id={null}
          >
            <svg
              fill="#0077DA"
              viewBox="0 0 32 32"
            >
              <path
                d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
              />
            </svg>
          </span>
          Call (415) 555-2323
        </a>
      </p>
      <p
        className="text-sm text-gray-700"
      >
        Due to high call volume you may hear a message.
      </p>
      <p
        className="my-3"
      >
        <a
          href="mailto:ae@email.com"
        >
          <span
            className="ui-icon ui-medium pr-2"
            data-test-id={null}
          >
            <svg
              fill="#0077DA"
              viewBox="0 0 32 32"
            >
              <path
                d="M29.091 27.637h-26.182c-1.606 0-2.909-1.304-2.909-2.909v-17.454c0-1.606 1.304-2.909 2.909-2.909h26.182c1.606 0 2.909 1.304 2.909 2.909v17.454c0 1.606-1.304 2.909-2.909 2.909zM27.637 24.728l-8.338-7.296-3.299 2.933-3.299-2.933-8.338 7.296h23.272zM2.909 23.272l8.182-7.272-8.182-7.272v14.546zM4.363 7.272l11.637 10.182 11.637-10.182h-23.272zM29.091 8.728l-8.182 7.272 8.182 7.272v-14.546z"
              />
            </svg>
          </span>
          Email
        </a>
      </p>
      <p
        className="text-gray-700 mb-1"
      >
        <span>
          1 South Van Ness Ave  
          <br />
          San Francisco , CA 94131
        </span>
      </p>
      <p
        className="mb-4"
      >
        <a
          className="inline-block pt-1"
          href="https://www.google.com/maps/place/1 South Van Ness Ave , San Francisco, CA 94131"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            className="ui-icon ui-medium pr-2"
            data-test-id={null}
          >
            <svg
              fill="#0077DA"
              viewBox="0 0 18 32"
            >
              <path
                d="M13.714 11.429q0-1.893-1.339-3.232t-3.232-1.339-3.232 1.339-1.339 3.232 1.339 3.232 3.232 1.339 3.232-1.339 1.339-3.232zM18.286 11.429q0 1.947-0.589 3.197l-6.501 13.821q-0.286 0.589-0.848 0.928t-1.205 0.339-1.205-0.339-0.83-0.928l-6.518-13.821q-0.589-1.25-0.589-3.197 0-3.786 2.678-6.464t6.464-2.678 6.464 2.678 2.678 6.464z"
              />
            </svg>
          </span>
          Get Directions
        </a>
      </p>
      <div
        className="my-3"
      >
        <h3
          className="text__caps-weighted"
        >
          Office Hours
        </h3>
        <div
          className="text-gray-800 text-tiny markdown"
        >
          <span>
            8-4 Mon-Thurs

            <br />
            9-4 Fri

            <br />
            10-2 Weekends
          </span>
        </div>
      </div>
    </section>
  </div>
  <div
    className="border-b border-gray-400 md:border-b-0 last:border-b-0"
  >
    <section
      className="aside-block "
    >
      <h4
        className="text__underline-weighted"
      >
        Housing Program
      </h4>
      <div
        className="text-tiny text-gray-750"
      >
        <a
          className="text-base"
          href="https://sfmohcd.org/for-buyers"
          target="_blank"
        >
          Below Market Rate
        </a>
      </div>
    </section>
  </div>
  <div
    className="border-b border-gray-400 md:border-b-0 last:border-b-0"
  >
    <section
      className="aside-block "
    >
      <div
        className="text-tiny text-gray-750"
      >
        <p>
          Listing Updated: April 11, 2022
        </p>
        <p
          className="mt-1"
        >
          <a
            className=""
            href="https://thefrontsteps.com/featured-listings/sfar/mls-481663-2820_scott_street_san_francisco_ca_94123/"
            target="_blank"
          >
            See this unit on MLS
          </a>
        </p>
      </div>
    </section>
  </div>
</li>
`;
